asyncpgsa/connection.py

```diff
@@ -15,32 +15,41 @@
 _dialect.paramstyle = 'named'
 
 
+def replace_keys(keys, querystring, params, inline=False):
+    new_query = querystring
+    new_params = []
+    for key, value in enumerate(keys):
+        if inline:
+            new_query = new_query.replace(':' + value, params[key][1], 1)
+        else:
+            new_query = new_query.replace(':' + value, '$' + str(key + 1), 1)
+            new_params.append(params[key][1])
+
+    return new_query, new_params
+
+
+def get_keys(compiled):
+    import re
+    pattern = ':(\w+)(?:\W|)'
+    p = re.compile(pattern, re.M)
+    keys = re.findall(p, compiled.string)
+    params = []
+    for i in keys:
```

Comment:
Could be a comprehension

```
params = [(i, compiled.params[i]) for i in keys]
```

Also, should probably handle the case where `i` isnt in the compiled.params list (they forgot to add one). Could just wrap this with a KeyError and raise a better more clear error